src/code_8057824.c

```diff
@@ -466,26 +522,1159 @@ s32 sub_8057CD0(struct DungeonEntity * pokemon, struct DungeonEntity * target, s
 bool8 sub_8057D20(struct DungeonEntity * pokemon, struct DungeonEntity * target, struct PokemonMove * move, u32 param_4)
```

Comment:
This is probably Whirlpool. It inflicts squeeze and does double damage to a target using Dive.